component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -63,10 +58,14 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
         if (context.isLogoutRequest()) {
             return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
         } else if (request.getParameter(TOTPAuthenticatorConstants.SEND_TOKEN) != null) {
-            if (generateTOTPToken(context)) {
-                return AuthenticatorFlowStatus.INCOMPLETE;
-            } else {
-                return AuthenticatorFlowStatus.FAIL_COMPLETED;
+            try {
+                if (generateTOTPToken(context)) {
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                } else {
+                    return AuthenticatorFlowStatus.FAIL_COMPLETED;
+                }
+            } catch (AuthenticationFailedException e) {
+                log.error("Error when generating the totp token");
```

Comment:
Why just log here. Also log does not have full stack trace. Don't we have to terminate the flow?